datalad/distributed/drop.py

```diff
@@ -727,6 +727,9 @@ def _kill_dataset(ds):
 def _drop_allkeys(ds, repo, force=False, jobs=None):
     """
     """
+    assert not (repo.dot_git / 'annex').is_symlink(), \
+        "Dropping from a symlinked annex is unsupported to prevent data-loss"
```

Comment:
In general I am good with such a stop gap feature (after testing is fixed up on crippled fs).  But also such message could confuse some naive user a little.  We might better rely first on

```
[datalad "clone"]
	reckless = ephemeral
```
being present in repo.config -- and inform user that dropping in ephemeral clone (the term they must have heard about since they had to use it to get here) is not supported.  But indeed checking for symlink is more robust and could be the 2nd line of defense if so desired.

Reply:
I disagree. Whether or not this config item is still set makes little difference re the outcome -- it remains data loss.

I don't understand what "2nd line of defense if so desired" wants to say.

Reply:
If I decode @yarikoptic's comment correctly, its about rewording the message under certain circumstances:
1: Check if the config item about an ephemeral clone
2: If it is present, reword the warning to talk about "ephemeral clone" instead of "symlinked annex"
The symlink check and warning would co-exist and catch a symlinked annex even without the ephemeral config.
Is that interpretation correct?

Reply:
For now let's just merge, and if @mih decides to improve UX, could be in a follow up PR.

Reply:
I will adjust the title to be little more "obvious" and reflective of the change.